hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -352,6 +356,10 @@ public void refreshTimeline() throws IOException {
       Option<Dataset<Row>> transformed =
           dataAndCheckpoint.getBatch().map(data -> transformer.get().apply(jssc, sparkSession, data, props));
       checkpointStr = dataAndCheckpoint.getCheckpointForNextBatch();
+      if(transformed.isPresent()){
```

Comment:
@vinothchandar : do you think we could do something like this. store the df schema and use it later when we initialize the writeclient config, where in we use this schema when target schema is null ? 
This is just WIP diff. haven't fully completed yet. but should give you an idea.